hr_employee_service/models/hr_employee.py

```diff
@@ -0,0 +1,93 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+from math import fabs
+from dateutil.relativedelta import relativedelta
+
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    service_hire_date = fields.Date(
+        string='Hire Date',
+        groups='hr.group_hr_user',
+        track_visibility='onchange',
+        help=(
+            'Hire date is normally the date an employee completes new hire'
+            ' paperwork'
+        ),
+    )
+    service_start_date = fields.Date(
+        string='Start Date',
+        groups='hr.group_hr_user',
+        track_visibility='onchange',
+        help=(
+            'Start date is the first day the employee actually works and'
+            ' this date is used for accrual leave allocations calculation'
+        ),
+    )
+    service_termination_date = fields.Date(
+        string='Termination Date',
+        groups='hr.group_hr_user',
+        track_visibility='onchange',
+        help=(
+            'Termination date is the last day the employee actually works and'
+            ' this date is used for accrual leave allocations calculation'
+        ),
+    )
+    service_duration = fields.Integer(
+        string='Service Duration',
+        readonly=True,
+        compute='_compute_service_duration',
+        help='Service duration in days',
+    )
+    service_duration_years = fields.Integer(
+        string='Service Duration (years)',
+        readonly=True,
+        compute='_compute_service_duration_display',
+    )
+    service_duration_months = fields.Integer(
+        string='Service Duration (months)',
+        readonly=True,
+        compute='_compute_service_duration_display',
+    )
+    service_duration_days = fields.Integer(
+        string='Service Duration (days)',
+        readonly=True,
+        compute='_compute_service_duration_display',
+    )
+
+    @api.multi
+    @api.depends('service_start_date', 'service_termination_date')
+    def _compute_service_duration(self):
+        for record in self:
+            if record.service_start_date:
+                service_since = record.service_start_date
+                service_until = (
+                    record.service_termination_date or fields.Date.today()
+                )
+                service_duration = fabs(
+                    (service_since - service_until).total_seconds()
+                )
+                record.service_duration = int(service_duration / 86400.0)
+            else:
+                record.service_duration = 0
+
+    @api.multi
+    @api.depends('service_start_date', 'service_termination_date')
+    def _compute_service_duration_display(self):
+        for record in self:
+            if record.service_duration:
+                service_duration = relativedelta(
+                    record.service_termination_date or fields.Date.today(),
+                    record.service_start_date
+                )
```

Comment:
I can see the reason why, but it's not nice to duplicate the service duration computation here.
I suggest this design:
- `_compute_service_duration` becomes a `_get_service_duration_timedelta` helper method, returning a timedelta (making it easy to extend the basic computation - I've seen tricky rules to handle 28/29/31 day months).
- the `_compute_service_duration` calls the `_get_service_duration_timedelta` helper and assigns value to all four computed fields.

Reply:
Got your idea! Yet not sure if it's better to proceed with `relativedelta` or with `timedelta`, what's your vision on that? First one would give super exact as far as I understand, yet does not provide the "total days" option

Reply:
IIIC timedelta is the object resulting from the relativedelta operation.
The point is that the helper function return a delta object, not usable directly by the ORM, but to be used by the actual _compute method.

Reply:
Timedelta has days and below (hours, minutes) and total_seconds(), while relativedelta gives exact years, months and days, the actual thing that is. So there's really to different calculations. I understand what you want to achieve yet still don't get how to accomplish that taking into account the different of calculation

Reply:
I see. 
Is it reasonable for you to compute `service_duration` from the years/months/days fields? 
If in years (as you originally had): = years + months/12 + days/360

Reply:
Or vice-versa as years * 365 + months * 30 + days
Personally, I'd keep as is since days don't loose any precision, leaving for other module consumers the decision how to use that data while those 3 extra fields are for UI purpose only